File: backend/openstream/schemas/channel.py
from pydantic import BaseModel, ConfigDict


class ChannelResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    stream_url: str

    tvg_id: str | None = None
    tvg_name: str | None = None

    logo_url: str | None = None

    group_title: str | None = None
    category: str | None = None

    country: str | None = None
    language: str | None = None

    playlist_id: int
    epg_channel_id: int | None = None


class PaginatedChannelResponse(BaseModel):
    items: list[ChannelResponse]

    page: int
    size: int
    total: int
    pages: int

File: frontend/src/types/channel.ts
export interface Channel {
  id: number;
  name: string;
  stream_url: string;
  tvg_id: string | null;
  tvg_name: string | null;
  logo_url: string | null;
  group_title: string | null;
  category: string | null;
  country: string | null;
  language: string | null;
  playlist_id: number;
  epg_channel_id: number | null;
}

export interface PaginatedChannels {
  items: Channel[];
  total: number;
  page: number;
  size: number;
}